### DISCUSSION TITLE
Property 'root' is protected and only accessible within class 'DefaultModelState' and its subclasses.

### DISCUSSION THREAD

**Discussion by csamarchi** [eclipse-glsp/glsp#687] [General]
I am following the Source Model structure for a Node server GLSP vscode extension. I have included all these [files](https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/node-json-vscode/glsp-server/src/model) verbatim.  I am getting the following error at this point in my code when building:

<img width="703" alt="Screen Shot 2022-06-21 at 3 25 48 PM" src="https://user-images.githubusercontent.com/42501494/174891207-455fb352-0c90-4532-a1d9-6dc59be91d6d.png">

`Property 'root' is protected and only accessible within class 'DefaultModelState' and its subclasses.`

I have tried replacing `protected` with `public` and still nothing.

I understand private/protected properties are only allowed within their class or subclass, but I am injecting the `ModuleModelState` part class within this file. I'm not sure how to move forward. Also unsure where "root" comes from exactly. 

Any help is much appreciated. Thanks

**Comment by tortmayr** [comment]
Hi @csamarchi,
The `ModelState` interface offers a dedicated [`updateModelRoot` method](https://github.com/eclipse-glsp/glsp-server-node/blob/b178eb2cf7509fdfa3349e012b33c651340dc0bb/packages/server-node/src/features/model/model-state.ts#L31) that you can use here:

```diff
-      this.modelState.root= GGraph.builder().id(module.id).addChildren(inputNodes).build();
+      this.modelState.updateRoot( GGraph.builder().id(module.id).addChildren(inputNodes).build());
```

If you are looking for an example please have a look at the [official documentation](https://www.eclipse.org/glsp/documentation/gmodel/).
There you can find a code snippet that demonstrates how to create your own custom `ModelFactory`:
```ts
@injectable()
export class MyModelFactory implements GModelFactory {
  @inject(MyModelState)
  protected modelState: MyModelState;

  createModel(): void {
    const entities = this.modelState.getModel().getEntities();

    const entityNodes = entities.map((entity) =>
      new GNodeBuilder(GNode)
        .id("node:entity")
        .layout("vbox")
        .add(new GLabelBuilder(GLabel).text(entity.name).build())
        .build()
    );

    const newModel = new GGraphBuilder(GGraph)
      .id("entity-graph")
      .addChildren(...entityNodes)
      .build();

    this.modelState.updateRoot(newModel);
  }
}```

**Comment by csamarchi** [comment]
Thank you! That fixed it. 

On the same subject of getting my Source Model to work, I am running into one more issue in my `storage.ts `file. I copied the example linked in the official documentation [here](https://github.com/eclipse-glsp/glsp-examples/blob/master/project-templates/node-json-vscode/glsp-server/src/model/tasklist-storage.ts). 

I am getting the error `error TS2740: Type '{ id: any; inputs: never[]; duration: number; taskType: string; name: string; references: string; description: string; source: string; }' is missing the following properties from type 'Module': type, [GLayoutContainer], position, size, and 6 more.`
 
On this section of code: 

<img width="566" alt="Screen Shot 2022-06-22 at 11 18 53 AM" src="https://user-images.githubusercontent.com/42501494/175082932-bd2fcc00-f47e-4726-b789-9fab66dc8cd9.png">

I have tried to specify all of the required properties on the object and tried marking the properties as optional with a '?' after the property. Both routes don't work. Do you have a solution?

Any help is appreciated. Thanks
